crates/autopilot/src/run_loop.rs

```diff
@@ -121,9 +124,68 @@ impl RunLoop {
         driver: &Driver,
         _solution: &solve::Response,
     ) -> Result<()> {
-        let request = execute::Request { auction_id: id };
+        let request = execute::Request {
+            auction_id: id,
+            tag: id.to_be_bytes().into(),
+        };
         let _response = driver.execute(&request).await.context("execute")?;
-        // TODO: Wait for transaction to be mined or deadline to be reached.
+        // TODO: React to deadline expiring.
+        self.wait_for_settlement_transaction(&request.tag)
+            .await
+            .context("wait for settlement transaction")?;
         Ok(())
     }
+
+    /// Tries to find a `settle` contract call with calldata ending in `tag`.
+    ///
+    /// Returns None if no transaction was found within the deadline.
+    pub async fn wait_for_settlement_transaction(&self, tag: &[u8]) -> Result<Option<Transaction>> {
+        // Start earlier than current block because there might be a delay when receiving the
+        // Solver's /execute response during which it already started broadcasting the tx.
```

Comment:
Wouldn't the proper solution be to ensure that you start "waiting for" or "expecting" the transaction _before_ calling `/execute`? Instead of going back some random X number of blocks.

Reply:
I don't think so because of the peer to peer to nature of the blockchain. With re-orgs and different delays of when blocks are observed it can be unpredictable for what block the transaction shows up. Maybe with proof of stake this can't happen or is extremely unlikely but in general I think we should have this buffer.